### PR TITLE
Prevent cadence from directly accessing fvm internal state

### PR DESCRIPTION
**Pull request by pattyshack**
Cadence shouldn't have access to fvm internal state via the value store.  The cadence runtime api as defined today allows arbitrary/unlimited access.

### CODE REVIEW

**Review by Tonix517 — APPROVED** [review]

**Review by janezpodhostnik — APPROVED** [review]

File: fvm/state/state.go [hunk 1]
```diff
@@ -318,39 +325,27 @@ func addressFromOwner(owner string) (flow.Address, bool) {
...
 // return false otherwise (key controlled by the cadence env)
-func IsFVMStateKey(owner, key string) bool {
-
+func IsFVMStateKey(owner string, key string) bool {
 	// check if is a service level key (owner is empty)
 	// cases:
 	// 		- "", "uuid"
 	// 		- "", "account_address_state"
-	if len(owner) == 0 {
+	if len(owner) == 0 && (key == UUIDKey || key == AddressStateKey) {
```
[the remark this answers is not in the document]
Comment (pattyshack) [reply fvm/state/state.go:335]: Are there any cases where FVM uses cadence to access its own keys (e.g., are meter weights consider FVM internal keys)?

**Comment by pattyshack** [comment]
bors merge

File: fvm/state/state.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (janezpodhostnik) [reply fvm/state/state.go:335]: meter weights are stored in the service account's storage, as a cadence controlled storage entry, so in this case the FVM needs cadence to fetch metering settings for the FVM. 

There are other instances where FVM settings are controlled by a value stored and managed by cadence:
- account creation allow/block list
- contract deploy allow/block list & settings
- ... (probably some other stuff I don't remember right now)

[the remark this answers is not in the document]
Comment (pattyshack) [reply fvm/state/state.go:335]: i have a noob question ... is the service account address the same as having no address owner (and if they are different, why)?

**Comment by pattyshack** [comment]
bors merge

**Pull request merged**